### PR TITLE
Doc ja 12 4 release

### PR DESCRIPTION
**Pull request by harukat**
12.4 リリースノート翻訳です。

### CODE REVIEW

**Review by KenichiroTanaka — COMMENTED** [review]
半分見ました。後ほど後半を見ます。

File: doc/src/sgml/release-12.sgml [hunk 1]
```diff
@@ -560,6 +742,11 @@ Branch: REL_12_STABLE [171633ff5] 2020-07-21 19:40:44 -0400
       where the query itself would work, so this means that we must use
       the query's collation when invoking operators for estimation
       purposes.
+-->
+これまでは、データベースの列に基づく照合順序が使われていました。
+しかし、問い合わせの照合順序を使う方が間違いなく適正です。
+さらに重要なことには、現在は非決定的照合順序があるため、非決定的照合順序が場合に演算子が与えられた際に完全に失敗する場合があります。
```
Comment (koizumistr) [inline doc/src/sgml/release-12.sgml:748]: 非決定的照合順序が場合に演算子が与えられた
→ 非決定的照合順序が演算子に与えられた
でしょうか。

File: doc/src/sgml/release-12.sgml [hunk 2]
```diff
@@ -619,15 +820,22 @@ Branch: REL_12_STABLE [28a862e9d] 2020-07-03 15:10:33 +0200
...
      </para>
 
      <para>
+<!--
       If <varname>temp_tablespaces</varname> is empty or explicitly names
       the database's primary tablespace, such files got placed into
       the <literal>pg_default</literal> tablespace rather than the
       database's primary tablespace as expected.
+-->
+<varname>temp_tablespaces</varname>が空あるいは明示的にデータベースの主テーブル空間が指定された場合に、このファイルは予期されるデータベースの主テーブルスペースではなく、<literal>pg_default</literal>テーブルスペースに置かれました。
```
Comment (koizumistr) [inline doc/src/sgml/release-12.sgml:838]: tablespaceの訳が「テーブルスペース」となっているところと「テーブル空間」となっているところがあります。
文書全体では「テーブル空間」が多数派のようです。

File: doc/src/sgml/release-12.sgml [hunk 3]
```diff
@@ -852,13 +1124,20 @@ Branch: REL_11_STABLE [70ae82b9b] 2020-05-16 18:16:37 +0900
 Branch: REL_10_STABLE [3acb30b49] 2020-05-16 18:16:41 +0900
 -->
      <para>
+<!--
       Fix assertion in logical replication subscriber to allow use
       of <literal>REPLICA IDENTITY FULL</literal> (Euler Taveira)
+-->
+ロジカルレプリケーションのサブスクライバで<literal>REPLICA IDENTITY FULL</literal>が使用できるようにアサートを修正しました。
```
Comment (koizumistr) [inline doc/src/sgml/release-12.sgml:1131]: サブスクライバーと延ばすのが多数派のようですが、どうなんでしょう。

**Review by koizumistr — COMMENTED** [review]
対応ありがとうございます。私の指摘部分については確認しました。

File: doc/src/sgml/release-12.sgml [hunk 4]
```diff
@@ -1174,16 +1553,23 @@ Branch: REL_11_STABLE [1785ac8ad] 2020-08-02 11:00:12 -0400
 Branch: REL_10_STABLE [e1b4135cf] 2020-08-02 11:00:12 -0400
 -->
      <para>
+<!--
       Fix corner case in decompression logic
       in <filename>contrib/pgcrypto</filename>'s
       <function>pgp_sym_decrypt</function> functions (Kyotaro Horiguchi,
       Michael Paquier)
+-->
+<filename>contrib/pgcrypto</filename>の<function>pgp_sym_decrypt</function>関数における圧縮の展開ロジックで稀な場合を修正しました。
```
Comment (KenichiroTanaka) [inline doc/src/sgml/release-12.sgml:1562]: 原文に記載されていないのが問題なのですが、稀にどうなるのかが分からないため補足が必要かと思います。

Author: Michael Paquier <michael@paquier.xyz>
Date:   Mon Jul 27 15:58:59 2020 +0900

    Fix corner case with 16kB-long decompression in pgcrypto, take 2
    
    A compressed stream may end with an empty packet.  In this case
    decompression finishes before reading the empty packet and the
    remaining stream packet causes a failure in reading the following
    data.  This commit makes sure to consume such extra data, avoiding a
    failure when decompression the data.  This corner case was reproducible
    easily with a data length of 16kB, and existed since e94dd6a.  A cheap
    regression test is added to cover this case based on a random,
    incompressible string.

上記がcommitログなので、

「<filename>contrib/pgcrypto</filename>の<function>pgp_sym_decrypt</function>関数における圧縮の展開ロジックで稀に不正なパケットを含めてしまう不具合を修正しました。」

とするのはどうでしょうか。

File: doc/src/sgml/release-12.sgml [hunk 5]
```diff
@@ -1174,16 +1553,23 @@ Branch: REL_11_STABLE [1785ac8ad] 2020-08-02 11:00:12 -0400
...
+(Kyotaro Horiguchi, Michael Paquier)
      </para>
 
      <para>
+<!--
       A compressed stream can validly end with an empty packet, but the
       decompressor failed to handle this and would complain about corrupt
       data.
+-->
+圧縮されたストリームは空のパケットで正当に終了する可能性がありますが、展開ロジックがこれを扱えず、破損データに対する苦情をあげていました。
```
Comment (KenichiroTanaka) [inline doc/src/sgml/release-12.sgml:1572]: ここも、
>破損データに対する苦情をあげていました。
→破損データのエラーを出力していました。

ではどうでしょうか。

**Review by KenichiroTanaka — CHANGES_REQUESTED** [review]
後半を確認しました。

File: doc/src/sgml/release-12.sgml (hunk 4 above)
Comment (harukat) [reply doc/src/sgml/release-12.sgml:1562]: 「場合を修正」は少し変なので「稀な場合の**不具合を**修正しました。」とします。
次パラグラフに説明がありますし、翻訳を逸脱するので、「不正なパケット・・・」を加えるのは賛同できません。

**Review by KenichiroTanaka — APPROVED** [review]
対応ありがとうございます。
一通り確認しました。
